website/src/components/Visualisations/IIGOPayments/Util/ProcessedPaymentsData.ts

```diff
@@ -0,0 +1,116 @@
+                switch (type) {
+                    case 'IslandAllocation':
+                        {
+                            const team =
+                                TeamName[
+                                    teamAction.ClientID as keyof typeof TeamName
+                                ]
+                            const current: TaxData | undefined = TaxInfo.get(
+                                team
+                            )
+                            if (current) {
+                                current.actualAlloc +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedAlloc +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    case 'IslandTaxContribution':
+                        {
+                            const team =
+                                TeamName[
+                                    teamAction.ClientID as keyof typeof TeamName
+                                ]
+                            const current: TaxData | undefined = TaxInfo.get(
+                                team
+                            )
+                            if (current) {
+                                current.actualTax +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedTax +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    case 'SanctionPaid':
+                        {
+                            const team =
+                                TeamName[
+                                    teamAction.ClientID as keyof typeof TeamName
+                                ]
+                            const current: TaxData | undefined = TaxInfo.get(
+                                team
+                            )
+                            if (teamAction.Pairs[0].Values[0] !== 0) {
+                                console.log(turnNumber, teamAction)
+                            }
+                            if (current) {
+                                current.actualSanction +=
+                                    teamAction.Pairs[0].Values[0]
+                                current.expectedSanction +=
+                                    teamAction.Pairs[1].Values[0]
+                                TaxInfo.set(team, current)
+                            }
+                        }
+                        break
+                    default:
+                        break
+                }
+            })
+        }
+    })
+    console.log(TaxInfo)
```

Comment:
can use `console.debug` :) (you'll have to turn on all log levels on your console however)